File: Solutions/2275. Largest Combination With Bitwise AND Greater Than Zero/2275. Largest Combination With Bitwise AND Greater Than Zero.go
func largestCombination(candidates []int) int {
    res:=0
    mx:=-1
    for _,c:=range candidates{mx=max(mx,c)}
    for b:=1;b<=mx;b*=2{
        cnt:=0
        for _,c:=range candidates{
            if c&b>0{
                cnt+=1   
            }
            res=max(res,cnt)
        }
    }
    return res
}

func max(i,j int) int{
    if i>j{
        return i
    }
    return j
}

File: Solutions/2275. Largest Combination With Bitwise AND Greater Than Zero/2275. Largest Combination With Bitwise AND Greater Than Zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        int res=0,mx=*max_element(candidates.begin(),candidates.end());
        for(int b=1;b<=mx;b*=2){
            int cnt=0;
            for(auto& c:candidates){
                cnt+=(c&b)>0;
            }
            res=max(res,cnt);
        }
        return res;
    }
};